app/models/user.rb

```diff
@@ -4,4 +4,21 @@ class User < ActiveRecord::Base
   validates :credit_card_form_url, url: {allow_blank: true, no_local: true, schemes: ['http', 'https']}
   validates :duns_number, duns_number: true
   validates :email, allow_blank: true, email: true
+
+  def from_oauth_hash(auth_hash)
+    set_if_blank('name', auth_hash)
+    set_if_blank('email', auth_hash)
+    self.github_login = auth_hash[:info][:nickname]
```

Comment:
note: `nickname` is github username https://github.com/intridea/omniauth/wiki/Auth-Hash-Schema (super confusing)